src/js/jsx/Toolbar.jsx

```diff
@@ -95,9 +98,15 @@ define(function (require, exports, module) {
 
         // On startup, we want to make sure center offsets take pinned toolbar
         componentDidMount: function () {
+            this._updateToolbarWidthDebounced = synchronization.debounce(this._updateToolbarWidth, this, 500);
+            os.addListener("displayConfigurationChanged", this._updateToolbarWidthDebounced);
```

Comment:
Whoopsie daisy: the toolbar was previously not updating it's dimensions at all on `displayConfigurationChange`.